src/biotite/sequence/profile.py

```diff
@@ -0,0 +1,244 @@
+# This source code is part of the Biotite package and is distributed
+# under the 3-Clause BSD License. Please see 'LICENSE.rst' for further
+# information.
+
+import numpy as np
+from .seqtypes import NucleotideSequence, ProteinSequence, GeneralSequence
+from .alphabet import LetterAlphabet
+from .align.alignment import get_codes
+
+__name__ = "biotite.sequence"
+__author__ = "Maximilian Greil"
+__all__ = ["SequenceProfile"]
+
+# Abbreviations
+_NUC_DNA_ALPH = NucleotideSequence.alphabet_unamb
+_NUC_RNA_ALPH = LetterAlphabet(["A", "C", "G", "U"])
+_PROT_ALPH = ProteinSequence.alphabet
+
+
+def _determine_common_alphabet(alphabets):
+    """
+    Determine the common alphabet from a list of alphabets, that
+    extends all alphabets.
+    """
+    common_alphabet = alphabets[0]
+    for alphabet in alphabets[1:]:
+        if not common_alphabet.extends(alphabet):
+            if alphabet.extends(common_alphabet):
+                common_alphabet = alphabet
+            else:
+                raise ValueError(
+                    "There is no common alphabet that extends all alphabets"
+                )
+    return common_alphabet
+
+
+def _codes_to_iupac(frequency, codes, maxes):
+    """
+    Returns IUPAC code for a row of 'symbols' with none, one or
+    multiple maximum positions.
+    """
+    if np.sum(frequency) == 0:
+        return "-"
+    key = ''.join(np.where(frequency == maxes)[0].astype(str))
```

Comment:
Can you use a tuple of integers instead of a string as dictionary key? This would remove the time-costly integer-to-string conversion. Furthermore, I think that a tuple of integers makes it more clear that they key is based on sequence codes.